remo/profiles/api/views.py

```diff
@@ -24,3 +45,179 @@ def retrieve(self, request, pk):
+        fields = ['country', 'category', 'initiative']
+
+
+class PeopleKPIView(APIView):
+
+    def get(self, request):
+        """Returns serialized data for People KPI."""
+
+        queryset = UserProfile.objects.filter(user__groups__name='Rep',
+                                              registration_complete=True)
+        people = PeopleKPIFilter(request.query_params, queryset=queryset)
+        weeks = int(request.query_params.get('weeks', KPI_WEEKS))
+
+        # Total number of Reps
+        total = people.count()
+
+        # Total Reps added in the last quarter
+        joined_date = get_quarter()[1]
+        quarter_total = people.qs.filter(
+            date_joined_program__gte=joined_date).count()
+
+        # Current quarter start
+        current_quarter_start = get_quarter()[1]
+
+        # Total Reps joined the previous quarter
+        previous_quarter_end = current_quarter_start - timedelta(days=1)
+        previous_quarter_start = get_quarter(previous_quarter_end)[1]
+        previous_quarter_total = people.qs.filter(
+            date_joined_program__range=[previous_quarter_start,
+                                        previous_quarter_end]).count()
+
+        diff = quarter_total - previous_quarter_total
+        try:
+            # Percentage change of events compared with previous week
+            quarter_ratio = diff/float(previous_quarter_total) * 100
+        except ZeroDivisionError:
+            if diff > 0:
+                quarter_ratio = 100
+            else:
+                quarter_ratio = 0
+
+        # Total Reps added this week
+        today = datetime.combine(now().date(), datetime.min.time())
+        current_week_start = today - timedelta(days=now().weekday())
+        prev_week_start = current_week_start - timedelta(weeks=1)
+
+        week_total = people.qs.filter(
+            date_joined_program__gte=current_week_start).count()
+
+        # Total Reps added the previous week
+        query_range = [prev_week_start, current_week_start]
+        prev_week_total = people.qs.filter(
+            date_joined_program__range=query_range).count()
+
+        diff = week_total - prev_week_total
+        try:
+            # Percentage change of events compared with previous week
+            week_ratio = diff/float(prev_week_total) * 100
+        except ZeroDivisionError:
+            if diff > 0:
+                week_ratio = 100
+            else:
+                week_ratio = 0
+
+        weekly_count = []
+        for i in range(weeks):
+            start = current_week_start - timedelta(weeks=i)
+            end = start + timedelta(weeks=1)
+
+            # Total number of reps (per week) for the past 6 weeks
+            count = people.qs.filter(
+                date_joined_program__range=[start, end]).count()
+            weekly_count.append({'week': weeks-i, 'people': count})
+
+        # Query params
+        q_kwargs = {'user__ng_reports__report_date__range': None}
```

Comment:
* ``Q()`` objects purpose is to encapsulate a collection of keyword arguments, so for us it make sense
* Using ``Q()`` object looks more clean and readable to me
* I don't really like the idea of using ``**kwargs`` when django provides ``Q()``